src/Symfony/Component/Security/Http/Attribute/IsGranted.php

```diff
@@ -21,12 +23,12 @@ public function __construct(
         /**
          * Sets the first argument that will be passed to isGranted().
          */
-        public array|string|null $attributes = null,
+        public array|string|Expression|null $attributes = null,
```

Comment:
@nicolas-grekas Just noticed this PR https://github.com/sensiolabs/SensioFrameworkExtraBundle/pull/648 , passing multiple attributes has been removed in #33584 . I've tested it and can confirm it doesn't work. As the PR mentions, cases like this should be handled with an expression. I'd suggest the following change:

```suggestion
        public string|Expression $attribute,
```

This would also remove the possibility of `#[IsGranted()]` not having an attribute at all since this doesn't seem to work either, nor can I find a reason why it even should. If I'm wrong about this one, please let me know.

I can do these changes on this PR or make a new one if that would make it easier to review.

Reply:
Good catch. A separate PR would be nice to help keep the focus in the discussion.

Reply:
Done, see #47144